### PR TITLE
Enhancement/mobile record functionality

### PR DESCRIPTION
**Pull request by vyasprateek**

### CODE REVIEW

File: Ginger/Ginger/Drivers/DriversWindows/MobileDriverWindow.xaml [hunk 1]
```diff
@@ -31,6 +37,21 @@
             <!--Top buttons-->
             <Frame x:Name="xMainTopFrame" Grid.Row="0" Grid.Column="1" Margin="0,0,0,0" Background="Black"/>
             <UserControls:ucButton x:Name="xRefreshButton"  Grid.Row="0" Grid.Column="1" ButtonType="ImageButton" HorizontalAlignment="Center" VerticalAlignment="Bottom" ButtonFontImageSize="18" ButtonImageType="Refresh" Click="xRefreshButton_Click" ToolTip="Refresh Device Screenshot" Margin="0,0,0,0" Visibility="Collapsed"/>
+            <UserControls:ImageMakerControl x:Name="xRecordingImage" Grid.Row="0" Grid.Column="1" Height="18" Width="18" ImageType="Camera" Foreground="Red" ImageForeground="Red" HorizontalAlignment="Right" VerticalAlignment="Center" SetBorder="False" ToolTip="Recording..." Margin="0,0,60,0" Visibility="Collapsed" />
+            <!--<UserControls:ImageMakerControl x:Name="xRecordingImage" Grid.Row="0" Grid.Column="1" Height="18" Width="18" ImageType="Camera" Foreground="Red" ImageForeground="Red" HorizontalAlignment="Right" VerticalAlignment="Center" SetBorder="False" ToolTip="Recording..." Margin="0,0,60,0" Visibility="Collapsed">
+                <Style>
+                    <Style.Triggers>
```
Comment (MeniKadosh1) [inline Ginger/Ginger/Drivers/DriversWindows/MobileDriverWindow.xaml:43]: Do cleanup for below, try to make the blinking recording work, if not working let me know and I'll try

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs [hunk 2]
```diff
@@ -62,6 +62,7 @@
 using System.Net;
 using System.Threading.Tasks;
 using System.Xml;
+using static GingerCore.Actions.ActMobileDevice;
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs:65]: why we having this static reference? not supposed to have it

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs [hunk 3]
```diff
@@ -1887,8 +1936,18 @@ void IRecord.ResetRecordingEventHandler()
...
+        bool IsRecording = false;
+
         void IRecord.StartRecording(bool learnAdditionalChanges)
         {
+            if(AppType == eAppType.Web)
+            {
+                mSeleniumDriver.StartRecording();
+            }
+            else
+                IsRecording = true;
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs:1948]: whyfor Web not to set **also** AppiumDriver.IsRecording= true?

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs [hunk 4]
```diff
@@ -2234,18 +2279,61 @@ public Byte[] GetScreenshotImage()
...
+            if (IsRecording)
+            {
+                ElementInfo elemInfo = GetElementAtPoint(x, y).Result;
+                if (elemInfo != null)
+                {
+                    RecordingEventArgs args = new RecordingEventArgs();
+                    args.EventType = eRecordingEvent.ElementRecorded;
+
+                    ElementActionCongifuration configArgs = new ElementActionCongifuration();
+                    configArgs.LocateBy = eLocateBy.ByXPath;
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs:2291]: need to change logic to set the locator by preference:
1- ID
2- Name
3- XPATH
4- X,Y

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs [hunk 5]
```diff
@@ -2669,6 +2757,31 @@ public override bool SerializationError(Agent agent, SerializationErrorType erro
         public void PerformScreenSwipe(eSwipeSide swipeSide, double impact = 1)
         {
             SwipeScreen(swipeSide, impact);
+
+            ActMobileDevice mobAct = new ActMobileDevice();
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs:2761]: all switch needs to be inside if (IsRecording) {} block no?

File: Ginger/GingerCoreNET/RecordingLib/RecordingManager.cs [hunk 6]
```diff
@@ -273,7 +273,7 @@ private void ElementRecordedHandler(ElementActionCongifuration args)
                 {
                     einfo = (ElementInfo)args.LearnedElementInfo;
                     args.AddPOMToAction = CreatePOM;
-                    args.POMGuid = CurrentPOM.Guid.ToString();
+                    args.POMGuid = CurrentPOM != null ? CurrentPOM.Guid.ToString() : string.Empty;
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/RecordingLib/RecordingManager.cs:276]: as disscused, needs to:
1- check how recording-POM integration working when there isn't any POM selected and how when there are multi POM selections
2- recommend on how it should behave in the difreant cases

**Review by MeniKadosh1 — CHANGES_REQUESTED** [review]

File: Ginger/Ginger/Drivers/DriversWindows/MobileDriverWindow.xaml (hunk 1 above)
Comment (vyasprateek) [reply Ginger/Ginger/Drivers/DriversWindows/MobileDriverWindow.xaml:37]: Done

**Review by MeniKadosh1 — APPROVED** [review]

**Pull request merged**